pavelib/paver_tests/test_database.py

```diff
@@ -59,3 +64,141 @@ def test_extract_files_from_zip(self, _mock_verify):
+
+
+def _write_temporary_db_cache_files(path, files):
+    """
+    create some temporary files to act as the local db cache files so that
+    we can compute a fingerprint
+    """
+    for index, filename in enumerate(files):
+        filepath = os.path.join(path, filename)
+        with open(filepath, 'w') as cache_file:
+            cache_file.write(str(index))
+
+
+class TestPaverDatabaseTasks(MockS3Mixin, TestCase):
+    """
+    Tests for the high level database tasks
+    """
+
+    def setUp(self):
+        super(TestPaverDatabaseTasks, self).setUp()
+        conn = boto.connect_s3()
+        conn.create_bucket('moto_test_bucket')
+        self.bucket = conn.get_bucket('moto_test_bucket')
+
+    @patch.object(db_utils, 'CACHE_FOLDER', mkdtemp())
+    @patch.object(db_utils, 'FINGERPRINT_FILEPATH', os.path.join(mkdtemp(), 'fingerprint'))
+    @patch.object(db_utils, 'sh')
+    def test_load_data_from_local_cache(self, _mock_sh):
+        """
+        Assuming that the computed db cache file fingerprint is the same as
+        the stored fingerprint, verify that we make a call to load data into
+        the database without running migrations
+        """
+        self.addCleanup(shutil.rmtree, db_utils.CACHE_FOLDER)
+        self.addCleanup(os.remove, db_utils.FINGERPRINT_FILEPATH)
+        _write_temporary_db_cache_files(db_utils.CACHE_FOLDER, database.ALL_DB_FILES)
+        # write the local fingerprint file with the same value than the
+        # computed fingerprint
+        expected_fingerprint = 'ccaa8d8dcc7d030cd6a6768db81f90d0ef976c3d'
+        with open(db_utils.FINGERPRINT_FILEPATH, 'w') as fingerprint_file:
+            fingerprint_file.write(expected_fingerprint)
+
+        database.update_local_bokchoy_db_from_s3()
+        calls = [
+            call('{}/scripts/calculate-bokchoy-migrations.sh'.format(Env.REPO_ROOT)),
+            call('{}/scripts/reset-test-db.sh'.format(Env.REPO_ROOT))
+        ]
+        _mock_sh.assert_has_calls(calls)
+
+    @patch.object(database, 'CACHE_BUCKET_NAME', 'moto_test_bucket')
+    @patch.object(db_utils, 'CACHE_FOLDER', mkdtemp())
+    @patch.object(db_utils, 'FINGERPRINT_FILEPATH', os.path.join(mkdtemp(), 'fingerprint'))
+    @patch.object(db_utils, 'sh')
+    def test_load_data_from_s3_fingerprint(self, _mock_sh):
+        """
+        Assuming that the computed db cache file fingerprint is different
+        than the stored fingerprint AND there is a matching fingerprint file
+        in s3, verify that we make a call to load data into the database
+        without running migrations
+        """
+        self.addCleanup(shutil.rmtree, db_utils.CACHE_FOLDER)
+        self.addCleanup(os.remove, db_utils.FINGERPRINT_FILEPATH)
+        _write_temporary_db_cache_files(db_utils.CACHE_FOLDER, database.ALL_DB_FILES)
+
+        # zip the temporary files and push them to a moto s3 bucket
+        expected_fingerprint = 'ccaa8d8dcc7d030cd6a6768db81f90d0ef976c3d'
```

Comment:
Nit: is this an arbitrary string? or the actual fingerprint for the temp files written with _write_temporary_db_cache_files?
  
If it's an arbitrary string, please leave that info in a comment.
If it's an actual fingerprint, I'm thinking that this test will start failing if for example, the number of files in the ALL_DB_FILES list changes. May be unimportant, we can fix at that time.